### DISCUSSION TITLE
Broadcast sending to Tower Collector app

### DISCUSSION THREAD

**Discussion by chipechop** [renyuneyun/Easer#431] [Q&A]
Hi, I'm trying to control with broadcast messages an application called Tower Collector, by Adam @Zamojski, with Easer, but I didn't succeed yet.

I want to: 
- stop Tower Collector, from collecting data (info.zamojski.soft.towercollector.COLLECTOR_STOP)
- start uploading data online (info.zamojski.soft.towercollector.UPLOADER_START)

![20220429_110805](https://user-images.githubusercontent.com/9009054/165922533-e1ac949e-86a7-4f29-9b5e-23cc4be80677.jpg)

First, I created an event at 3:00AM, and a profile with two actions (stop collecting, start uploading), **using the "Launch App" operation**, but it didn't work.

![Screenshot_20220429-113900_Easer (beta)](https://user-images.githubusercontent.com/9009054/165922641-afaa3860-98e5-4f4b-aab5-1812f213461b.jpg)

Then I modified the profile created above and used the **"Send Broadcast" operation**, instead of the the "Launch App" operation, but I don't understand how I should fill the fields of the configuration window. I tried what you can see in the picture, but it doesn't work...

![Screenshot_20220429-112916_Easer (beta)](https://user-images.githubusercontent.com/9009054/165922767-dd4ce7b8-52be-47c7-b28f-80c7f43e0482.jpg)

What should I write into the " Action, Category, Data, Type" fields?

Any help, appreciated.

**Comment by renyuneyun** [comment]
Hi. I think what you put in is correct based on your description. Sorry, I don't speak the language in the screenshot (is that Italian?) so can't help debug directly.

The "Action, Category, Data, Type" fields are all standard fields for an `Intent` (the core content of a broadcast) in Android. You'd better refer to the official Android document or other introduction materials if you want to understand (e.g. what they are, how they relate to each other).
I'll give a little more details to help you debug:
1. The values of these fields should be found from Tower Collector (TC) app. They should be defined by the author of TC, and used by TC. In general, TC should be responsible to document the necessary fields.
2. Normally, apps will specify at least an Action. Sometimes, they may also specify a Category (e.g. if there are so many Actions they may listen to). Android relies on these two fields to find the correct `BroadcastReceiver`, by exactly matching them. So you need at least ensure the values of them are correct so the broadcast can be received by TC (then interpreted by TC).

**Comment by zamojski** [comment]
It seems that Easer is missing the required field. Take a look on the screenshot from Automagic app 
![Screenshot_20220501-221046](https://user-images.githubusercontent.com/12007817/166163032-9b030463-1668-4d18-bdd2-08e011f0c5f0.jpg)

**Comment by OliveiraHermogenes** [comment]
As @zamojski pointed out, the "Send Broadcast" profile in Easer seems to be missing a field for the target package. With the "Execute command" profile, the following works: `am broadcast --user 0 -a info.zamojski.soft.towercollector.COLLECTOR_START info.zamojski.soft.towercollector`. Possibly related to #330. I am not familiar with Android internals, but perhaps a requirement introduced in API 30+? LineageOS 18.1 (Android 11) here, no root.
